extensions/amp-ad-network-doubleclick-impl/0.1/amp-ad-network-doubleclick-impl.js

```diff
@@ -1193,15 +1193,19 @@ export class AmpAdNetworkDoubleclickImpl extends AmpA4A {
   handleResize_(width, height) {
     const pWidth = this.element.getAttribute('width');
     const pHeight = this.element.getAttribute('height');
-    // We want to resize only if neither returned dimension is larger than its
-    // primary counterpart, and if at least one of the returned dimensions
-    // differ from its primary counterpart.
-    if (
-      (this.isFluidRequest_ && width && height) ||
-      ((width != pWidth || height != pHeight) &&
-        (width <= pWidth && height <= pHeight))
-    ) {
-      this.attemptChangeSize(height, width).catch(() => {});
+    const isFluidRequestAndFixedResponse =
+        !!(this.isFluidRequest_ && width && height);
+    const returnedSizeDifferent = width != pWidth || height != pHeight;
+    const heightNotIncreased = height <= pHeight
+    if (isFluidRequestAndFixedResponse ||
+        returnedSizeDifferent && heightNotIncreased) {
+      if (height == pHeight) {
+        // If we're only changing the width, then we call changeSize which
+        // should allow us to resize the element even if it's in the viewport.
+        this.element.getResources().changeSize(this.element, height, width);
```

Comment:
@dvoytenko @jridgewell is it intended to not allow width-only in-viewport resize in `attemptChangeSize`?

Reply:
I don't think we ever considered it... I'm personally fine with it. I'll let @dvoytenko decide.

Note, this needs to happen in a vsync mutate phase.

Reply:
@dvoytenko made a good point that width change can cause a reflow. for example, when a full width div reduce size to half, the empty space to its right can be filled by elements originally below it.

Reply:
@lannka  For the case where we want to modify the width only, we only expect it to grow larger. If I enforce that condition and put this inside a vsync mutate phase, would that be alright?

Reply:
It's the same for width increase. Imagine a half width div grows to full width, it could push the element right to it (if it has one) to the next row.

Reply:
Removed the controversial lines. Let me know if there's any further issue.